autofaiss/indices/distributed.py

```diff
@@ -336,3 +342,315 @@ def run(
+        min_nearest_neighbors_to_retrieve=min_nearest_neighbors_to_retrieve,
+        use_gpu=use_gpu,
+        make_direct_map=make_direct_map,
+    )
+
+
+def create_big_index(
+    embedding_root_dir: str,
+    ss,
+    output_root_dir: str,
+    id_columns: Optional[List[str]],
+    should_be_memory_mappable: bool,
+    max_index_query_time_ms: float,
+    max_index_memory_usage: str,
+    min_nearest_neighbors_to_retrieve: int,
+    embedding_column_name: str,
+    current_memory_available: str,
+    nb_cores: Optional[int],
+    use_gpu: bool,
+    metric_type: str,
+    nb_splits_per_big_index: int,
+    make_direct_map: bool,
+    temp_root_dir: str,
+) -> Optional[Dict[str, str]]:
+    """
+    Create a big index
+    """
+
+    def _create_and_train_index_from_embedding_dir() -> TrainedIndex:
+        trained_index = create_and_train_index_from_embedding_dir(
+            embedding_root_dir=embedding_root_dir,
+            embedding_column_name=embedding_column_name,
+            max_index_memory_usage=max_index_memory_usage,
+            make_direct_map=make_direct_map,
+            should_be_memory_mappable=should_be_memory_mappable,
+            use_gpu=use_gpu,
+            metric_type=metric_type,
+            nb_cores=nb_cores,
+            current_memory_available=current_memory_available,
+            id_columns=id_columns,
+        )
+
+        index_output_root_dir = os.path.join(temp_root_dir, "training", partition)
+        output_index_path = save_index(trained_index.index_or_path, index_output_root_dir, "trained_index")
+        return TrainedIndex(output_index_path, trained_index.index_key, embedding_root_dir)
+
+    partition = extract_partition_name_from_path(embedding_root_dir)
+
+    # Train index
+    rdd = ss.sparkContext.parallelize([embedding_root_dir], 1)
+    trained_index_path, trained_index_key, _, = rdd.map(
+        lambda _: _create_and_train_index_from_embedding_dir()
+    ).collect()[0]
+
+    # Add embeddings to index and compute metrics
+    partition_temp_root_dir = os.path.join(temp_root_dir, "add_embeddings", partition)
+    index, metrics = _add_embeddings_from_dir_to_index(
+        add_embeddings_fn=partial(
+            add_embeddings_to_index_distributed,
+            trained_index_or_path=trained_index_path,
+            nb_cores=nb_cores,
+            temporary_indices_folder=partition_temp_root_dir,
+            nb_indices_to_keep=nb_splits_per_big_index,
+        ),
+        embedding_root_dir=embedding_root_dir,
+        output_root_dir=output_root_dir,
+        index_key=trained_index_key,
+        embedding_column_name=embedding_column_name,
+        current_memory_available=current_memory_available,
+        id_columns=id_columns,
+        max_index_query_time_ms=max_index_query_time_ms,
+        min_nearest_neighbors_to_retrieve=min_nearest_neighbors_to_retrieve,
+        use_gpu=use_gpu,
+        make_direct_map=make_direct_map,
+    )
+
+    if index:
```

Comment:
Can you add a comment to explain why it is necessary/useful?

Reply:
Why do we need to del the object? it would have been deleted automatically at the return of the function